qiskit/circuit/classical/expr/expr.py

```diff
@@ -115,38 +115,57 @@ class Var(Expr):
     associated name; and an old-style variable that wraps a :class:`.Clbit` or
     :class:`.ClassicalRegister` instance that is owned by some containing circuit.  In general,
     construction of variables for use in programs should use :meth:`Var.new` or
-    :meth:`.QuantumCircuit.add_var`."""
+    :meth:`.QuantumCircuit.add_var`.
+
+    Variables are immutable after construction, so they can be used as dictionary keys."""
 
     __slots__ = ("var", "name")
 
+    var: qiskit.circuit.Clbit | qiskit.circuit.ClassicalRegister | uuid.UUID
```

Comment:
Python treats this as a pure type hint attached to the class, and iirc, this is the way to make Sphinx easily pick up both the docstring and the type hint (but I don't remember _exactly_ why I moved this).

A classvar isn't created because there's no assignment, and the type hint for a classvar that isn't created by the same line would be `var: ClassVar[str]` (or whatever).